### PR TITLE
Update to PHP 8.1

### LINKED ISSUE #221: Upgrade to PHP 8.1
We should jump on the wagon to use the soon to be released PHP 8.1.

### PR DESCRIPTION
**Pull request by Nispeon**
Updated the stack to use the brand new version of PHP!

Fix #221

### CODE REVIEW

**Review by lyrixx — APPROVED** [review]

File: src/Application/DiscordApplication.php [hunk 1]
```diff
@@ -31,20 +31,15 @@ class DiscordApplication implements ApplicationInterface
...
-    {
-        $this->requestStack = $requestStack;
-        $this->apiHelper = $apiHelper;
-        $this->userExtractor = $userExtractor;
-        $this->messageSender = $messageSender;
+    public function __construct(
+        private RequestStack $requestStack,
+        private ApiHelper $apiHelper,
+        private UserExtractor $userExtractor,
+        private MessageSender $messageSender
```
Comment (lyrixx) [inline src/Application/DiscordApplication.php:41]: Could you add a trailing `,` for each controller? 
Like that, future diff will be cleaner

File: src/Controller/SantaController.php [hunk 2]
```diff
@@ -12,6 +12,7 @@
 namespace JoliCode\SecretSanta\Controller;
 
 use Bugsnag\Client;
+use Iterator;
```
Comment (lyrixx) [inline src/Controller/SantaController.php:15]: We don't import root namespassed class

File: src/Exception/AuthenticationException.php [hunk 3]
```diff
@@ -13,12 +13,11 @@
...
-    {
-        $this->applicationCode = $applicationCode;
-
+    public function __construct(
+        private string $applicationCode,
+        string $message = '',
+        \Throwable $previous = null
+    ) {
         parent::__construct($message, 0, $previous);
     }
```
Comment (lyrixx) [inline src/Exception/AuthenticationException.php:22]: I'm not super fan of mixing class property and argument in the constructor.
But don't change it now, let's wait for more feedback

File: src/Model/Group.php [hunk 4]
```diff
@@ -13,15 +13,13 @@
...
+    private array $userIds = [];
 
-    public function __construct(string $identifier, string $name)
-    {
-        $this->identifier = $identifier;
-        $this->name = $name;
+    public function __construct(
+        private string $identifier,
+        private string $name
+    ) {
```
Comment (lyrixx) [inline src/Model/Group.php:22]: Same question for such cases

File: src/Application/DiscordApplication.php (hunk 1 above)
Comment (Nispeon) [reply src/Application/DiscordApplication.php]: Sure 😄

File: src/Controller/SantaController.php (hunk 2 above)
Comment (Nispeon) [reply src/Controller/SantaController.php]: My mistake ^^

File: src/Model/Group.php (hunk 4 above)
Comment (Nispeon) [reply src/Model/Group.php]: For the other class, I didn't make it that way so I don't know why it is like that.
But for this case, I believe I just changed the way of instantiating the construct parameters to the new PHP8.0 way, or did I fail somewhere ?

File: src/Discord/ApiHelper.php [hunk 5]
```diff
@@ -18,12 +18,11 @@
 
 class ApiHelper
 {
-    private const TOKEN_TYPE_BOT = 'Bot';
+    const TOKEN_TYPE_BOT = 'Bot';
 
-    private $botToken;
+    private string $botToken;
```
Comment (pyrech) [inline src/Discord/ApiHelper.php:23]: Move this props declaration to the constructor

**Review by pyrech — CHANGES_REQUESTED** [review]

File: src/Exception/AuthenticationException.php (hunk 3 above)
Comment (pyrech) [reply src/Exception/AuthenticationException.php]: Not a big deal IMO. I think I prefer consistency between all property declarations

**Review by pyrech — APPROVED** [review]

**Comment by damienalexandre** [comment]
Nice, thanks a lot!

**Pull request merged**